server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/MapStringTermsAggregator.java

```diff
@@ -117,6 +118,8 @@ public InternalAggregation buildEmptyAggregation() {
     public void collectDebugInfo(BiConsumer<String, Object> add) {
         super.collectDebugInfo(add);
         add.accept("total_buckets", bucketOrds.size());
+        add.accept("collection_strategy", collectorSource.describe());
+        collectorSource.collectDebugInfo(add);
```

Comment:
Without describing the collection strategy it isn't obvious from reading the profile that you are looking at `significant_text` rather than `significant_terms`. The extra debugging information should be useful in figuring out what makes a particular `significant_text` execution slow - maybe its fetching from source. Maybe its analysis. Maybe its just generating a zillion terms. That's all there in the debug output.